src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPOpenIdAuthenticator.java

```diff
@@ -0,0 +1,328 @@
+                    );
+                }
+
+                JWTClaimsSet claims;
+                boolean isSigned = contentType.contains(APPLICATION_JWT);
+                if (contentType.contains(APPLICATION_JWT)) { // We don't need the userinfo_encrypted_response_alg since the
+                                                             // selfRefreshingKeyProvider has access to the keys
+                    claims = openIdJwtAuthenticator.getJwtClaimsSetFromInfoContent(userinfoContent);
+                } else {
+                    claims = JWTClaimsSet.parse(userinfoContent);
+                }
+
+                String id = openIdJwtAuthenticator.getJwtClaimsSet(request).getSubject();
+                String missing = validateResponseClaims(claims, id, isSigned);
+                if (!missing.isBlank()) {
+                    throw new AuthenticatorUnavailableException(
+                        "Error while getting " + this.userinfo_endpoint + ": Missing or invalid required claims in response: " + missing
+                    );
+                }
+
+                final String subject = openIdJwtAuthenticator.extractSubject(claims);
+                if (subject == null) {
+                    log.error("No subject found in JWT token");
+                    return null;
+                }
+
+                final String[] roles = openIdJwtAuthenticator.extractRoles(claims);
+
+                AuthCredentials ac = new AuthCredentials(subject, roles);
+
+                for (Map.Entry<String, Object> claim : claims.getClaims().entrySet()) {
+                    ac.addAttribute("attr.jwt." + claim.getKey(), String.valueOf(claim.getValue()));
+                }
+
+                return ac.markComplete();
+            } catch (ParseException e) {
+                throw new RuntimeException(e);
+            }
+        } catch (IOException e) {
+            throw new AuthenticatorUnavailableException("Error while getting " + this.userinfo_endpoint + ": " + e, e);
+        }
+    }
+
+    private String validateResponseClaims(JWTClaimsSet claims, String id, boolean isSigned) {
+
+        String missing = "";
+
+        if (claims.getClaim(SUB_CLAIM) == null || claims.getClaim(SUB_CLAIM).toString().isBlank() || !claims.getClaim("sub").equals(id)) {
+            missing = missing.concat(SUB_CLAIM);
+        }
+
+        if (isSigned) {
+            if (claims.getIssuer() == null || claims.getIssuer().isBlank() || !claims.getIssuer().equals(settings.get(ISSUER_ID_URL))) {
+                missing = missing.concat("iss");
+            }
+            if (claims.getAudience() == null
+                || claims.getAudience().toString().isBlank()
+                || !claims.getAudience().contains(settings.get(CLIENT_ID))) {
+                missing = missing.concat("aud");
+            }
+        }
+
+        return missing;
+    }
+
+    private final class HTTPJwtKeyByOpenIdConnectAuthenticator extends AbstractHTTPJwtAuthenticator {
```

Comment:
All above is new. This internal class previously existed as the separate HttpJwtKeyByOpenIdConnectAuthenticator. I nested the class since the userinfo logic directly relies on it and the previous naming was out of convention.